### PR TITLE
Bluetooth: controller: fixing slot reservation for Central and Peripheral

### LINKED ISSUE #48823: Bluetooth: controller: llcp: limited nr. of simultaneous connections
**Describe the bug**
the babblesim test tests/bluetooth/bsim_bt/bsim_test_multiple fails if the maximum number of connections is higher than circa 170.
The test should pass with a maximum number of connections set to 250

**To Reproduce**
1. change the prj.conf file, so that CONFIG_BT_MAX_CONN=250
2. run the script 'compile.sh' or alternatively only compile the bsim_test_multiple sample
3. run the script tests/bluetooth/bsim_bt/bsim_test_multiple/test_scripts/multiple.sh
4. there will be error-messages saying that connecting failed after about 170 succesfull connections

Possible cause might be related to the allocated buffers, using an uint8_t instead of uint16_t causing memory corruption

### PR DESCRIPTION
**Pull request by erbr-ot**
With introduction of refactored LLCP was also introduced an error
re. calculation of initial slot reservation in connection setups. 

Due to this error a bigger slot was reserved, leading to failure due to
connection overlaps in multiple.sh test. 
Also the removal of this 'overhead' in slot reservations calls for
bigger time-allowance for l2cap_stress test. 

fixes #48823

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

### CODE REVIEW

**Review by cvinayak — APPROVED** [review]

File: tests/bluetooth/bsim_bt/bsim_test_l2cap_stress/src/common.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ extern enum bst_result_t bst_result;
 	}
 
 
-#define WAIT_SECONDS 120                         /* seconds */
```
Comment (jori-nordic) [inline tests/bluetooth/bsim_bt/bsim_test_l2cap_stress/src/common.h:39]: wow is it really that much slower, or did you add a fat margin?

**Review by jori-nordic — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@carlescufi, should this be merged for 3.2?

File: tests/bluetooth/bsim_bt/bsim_test_l2cap_stress/src/common.h (hunk 1 above)
Comment (erbr-ot) [reply tests/bluetooth/bsim_bt/bsim_test_l2cap_stress/src/common.h:39]: Well, yeah it is - no margin. 
The issue is that the way the connections get laid out they now 'collide' with only minimal slot reserved. Before the default (faulty) slot reservation left plenty of space in the slots to send/receive.

**Comment by kruithofa** [comment]
fixes #48823

**Comment by kruithofa** [comment]
> @carlescufi, should this be merged for 3.2?

@mmahadevan108 yes, please merge this for 3.2

**Review by kruithofa — APPROVED** [review]
LGTM

**Pull request merged**